mlflow/tracking/fluent.py

```diff
@@ -3396,26 +3404,50 @@ def _get_active_model_context() -> ActiveModelContext:
 
 def get_active_model_id() -> Optional[str]:
     """
-    Get the active model ID. If no active model is set with ``set_active_model()``, this will
-    try to get the model ID from the environment variable ``MLFLOW_ACTIVE_MODEL_ID``.
-    If neither is set, return None.
+    Get the active model ID. If no active model is set with ``set_active_model()``, the
+    default active model is set using model ID from the environment variable
+    ``_MLFLOW_ACTIVE_MODEL_ID``. If neither is set, return None.
+    Note that this function only get the active model ID from the current thread.
 
     Returns:
         The active model ID if set, otherwise None.
     """
     return _get_active_model_context().model_id
 
 
-def _reset_active_model_context() -> None:
+def _get_active_model_id_global() -> Optional[str]:
     """
-    Should be called only for testing purposes.
+    Get the active model ID from the global context by checking all threads.
+    This is useful when we need to get the active_model_id set by a different thread.
     """
-    _ACTIVE_MODEL_CONTEXT.set(ActiveModelContext())
-    _update_active_model_id_env_var(None)
+    # if the active model ID is set in the current thread, always use it
+    if model_id_in_current_thread := get_active_model_id():
+        _logger.info(f"Active model ID found in the current thread: {model_id_in_current_thread}")
+        return model_id_in_current_thread
+    model_ids = [
+        ctx.model_id
+        for ctx in _ACTIVE_MODEL_CONTEXT.get_all_thread_values().values()
+        if ctx.model_id is not None
+    ]
+    if model_ids:
+        if len(set(model_ids)) > 1:
+            _logger.warning(
+                "Failed to get one active model id from all threads, multiple active model IDs "
+                f"found: {set(model_ids)}."
+            )
+            return
+        return model_ids[0]
+    elif _MLFLOW_DEBUG.get():
+        _logger.warning("No active model ID found in any thread.")
 
 
-def _update_active_model_id_env_var(value):
-    if value is None:
-        MLFLOW_ACTIVE_MODEL_ID.unset()
-    else:
-        MLFLOW_ACTIVE_MODEL_ID.set(value)
+def unset_active_model() -> None:
+    """
+    Unset the active model. This will remove the active model previously set by
+    :py:func:`mlflow.set_active_model` from current thread. To temporarily switch
+    the active model, use ``with mlflow.set_active_model(...)`` instead.
+    """
+    # reset the environment variable as well to avoid it being used when creating
+    # ActiveModelContext
+    _MLFLOW_ACTIVE_MODEL_ID.unset()
+    _ACTIVE_MODEL_CONTEXT.set(ActiveModelContext())
```

Comment:
Question: after user calls this API, do we still want to link traces to a loaded model?
```
mlflow.unset_active_model()

model = mlflow.pyfunc.load_model(...)
model.predict(...) --> traces generated here
```

Reply:
> load_model() should set the active model

Make sense, then we can just clear the state.

> IMO, it makes sense for load_model() to call set_active_model().

We don't call set_active_model inside load_model because there could be a corner case like: 1. user sets an active model directly with `set_active_model`; 2. user loads a model with `load_model`. Then we might not want to change user's expectation implicitly, since they might load a model inside the model they're testing, or they just want to link all traces to one model_id.

Reply:
Can't the user swap the order if they want to achieve that? Seems like a pretty rare corner case. I'd rather have `load_model()` do `set_active_model()`.

Reply:
Currently load_model calls _set_active_model with the flag `set_by_user=False`, that's the only difference with set_active_model. Do you have concerns about current implementation?